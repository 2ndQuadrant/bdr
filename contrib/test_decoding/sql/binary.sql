CREATE EXTENSION test_decoding;
-- predictability
SET synchronous_commit = on;

SELECT 'init' FROM pg_create_decoding_replication_slot('regression_slot', 'test_decoding');
-- succeeds, textual plugin, textual consumer
SELECT data FROM pg_decoding_slot_get_changes('regression_slot', 'now', 'force-binary', '0');
-- fails, binary plugin, textual consumer
SELECT data FROM pg_decoding_slot_get_changes('regression_slot', 'now', 'force-binary', '1');
-- succeeds, textual plugin, binary consumer
SELECT data FROM pg_decoding_slot_get_binary_changes('regression_slot', 'now', 'force-binary', '0');
-- succeeds, binary plugin, binary consumer
SELECT data FROM pg_decoding_slot_get_binary_changes('regression_slot', 'now', 'force-binary', '1');
