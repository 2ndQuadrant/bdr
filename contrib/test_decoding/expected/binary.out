CREATE EXTENSION test_decoding;
ERROR:  extension "test_decoding" already exists
-- predictability
SET synchronous_commit = on;
SELECT 'init' FROM pg_create_decoding_replication_slot('regression_slot', 'test_decoding');
 ?column? 
----------
 init
(1 row)

-- succeeds, textual plugin, textual consumer
SELECT data FROM pg_decoding_slot_get_changes('regression_slot', 'now', 'force-binary', '0');
 data 
------
(0 rows)

-- fails, binary plugin, textual consumer
SELECT data FROM pg_decoding_slot_get_changes('regression_slot', 'now', 'force-binary', '1');
ERROR:  output plugin uses binary output, but a textual interface is used
-- succeeds, textual plugin, binary consumer
SELECT data FROM pg_decoding_slot_get_binary_changes('regression_slot', 'now', 'force-binary', '0');
 data 
------
(0 rows)

-- succeeds, binary plugin, binary consumer
SELECT data FROM pg_decoding_slot_get_binary_changes('regression_slot', 'now', 'force-binary', '1');
 data 
------
(0 rows)

